.github/workflows/abi-diff.yml

```diff
@@ -87,57 +96,72 @@ jobs:
         with:
           fetch-depth: 0  # needed to check out all branches for this Action to work
 
-      - name: Run diff
-        id: diff
-        run: |
-          # this is a workaround, if I put the github expansions directly into
-          # the line below these two it doesn't work
-          base_branch="${{ inputs.base-branch }}"
-          search_path="${{ inputs.search-path }}"
-          
-          diff_lines=$(git diff "remotes/origin/$base_branch" -- $search_path | wc -l)
-          echo "Diff has $diff_lines lines"
-          echo "nlines=$diff_lines" >> "$GITHUB_OUTPUT"
-
       - name: Get dependency provider
         uses: dice-group/cpp-conan-release-reusable-workflow/.github/actions/add_conan_provider@main
 
-      # Run cmake because the version file might be generated by it
-      - name: Configure CMake
-        env:
-          CC: ${{ steps.install_cc.outputs.cc }}
-          CXX: ${{ steps.install_cc.outputs.cxx }}
-        run: |
-          cmake -G Ninja -B build_dir -DCMAKE_PROJECT_TOP_LEVEL_INCLUDES=conan_provider.cmake
-
       - name: Create ABI version detection program
         run: |
           PROG=$(cat << EOF
           #include <iostream>
           #include <${{ inputs.abi-version-header }}>
-          
+
           int main() {
             std::cout << ${{ inputs.abi-version-const }} << std::endl;
           }
           EOF
           )
-          
+
           echo "$PROG" > /tmp/detect-abi-version.cpp
 
+      # Diff this branch vs the base branch
+      - name: Run diff
+        id: diff
+        run: |
+          # this is a workaround, if I put the github expansions directly into
+          # the line below these two it doesn't work
+          base_branch="${{ inputs.base-branch }}"
+          search_path="${{ inputs.search-path }}"
+          
+          diff_lines=$(git diff "remotes/origin/$base_branch" -- $search_path | wc -l)
+          echo "Diff has $diff_lines lines"
+          echo "nlines=$diff_lines" >> "$GITHUB_OUTPUT"
+
+      # TODO point to main
```

Comment:
The stupid thing with workflows is that you always have to specify a branch, so I basically need to break it by changing it to main and then immediately merge this PR into main to unbreak it. Idk if there is a better way to do this

Reply:
I think there is no better way. It's self-referential hell :)